### PR TITLE
Feature: `CustomAsyncImage` component

### PR DESCRIPTION
**Pull request by fespinoza**
After seeing how the `ViewModels` within the main app get complicated by having to fetch all nested structure's images
to be able to create the `ViewData` values I decided to clean up and refine the `CustomAsyncImage`[^1] I showed you before.

This implementation was inspired by the iOS 15 `AsyncImage` component.

You use it like this

```swift
CustomAsyncImage(.remote(url: url)) { image in
     image
         .resizable()
         .aspectRatio(contentMode: .fit)
         .frame(width: 200, height: 200)
         .background(Color.red)
 }
```

The "trick" is that the value for this component is a value of type: `ImageViewData`, which has the following cases

```swift
public enum ImageViewData: Equatable {
    case empty
    case remote(url: URL)
    case loading
    case image(_ image: Image)
}
```

So we can represent URLs as async images through `ImageViewData` and the component will be responsible for actually loading the image.

## Library initalization

Previously we used `SATSFont.registerCustomFonts()` on `AppDelegate` to make sure the library had the fonts it needed to work correctly. Since we need a `ImageClient` instance for the implementation details of `CustomAsyncImage`, I created a new `Config` class that needs to be initialized instead.

```
import SATSCore

// In AppDelegate
Config.setup()
```

This will register the fonts and accept a custom `ImageClient` if we need to.

## ⚠️ Notice

I recommend you to review this per-commit, I added some explanation about the use of `@ObservedObject` instead of `@StateObject` in the relevant commit.

[^1]: If you have a better name for this component, I'm open for that.

### CODE REVIEW

File: Sources/SATSCore/BasicComponents/CustomAsyncImage/ImageClient.swift [hunk 1]
```diff
@@ -13,3 +13,39 @@ public protocol ImageClient {
...
+        task?.resume()
+    }
+
+    enum ImageError: LocalizedError {
+        case decodeError
+        case noData
+
+        var errorDescription: String? {
+            switch self {
+            case .decodeError: return "We got the image data but coudn't convert it to a proper image"
```
Comment (kentrh) [inline Sources/SATSCore/BasicComponents/CustomAsyncImage/ImageClient.swift:46]: Typo coudn't -> couldn't

**Review by kentrh — APPROVED** [review]
Looking good! This will be super useful, thank you!

File: Sources/SATSCore/BasicComponents/CustomAsyncImage/ImageViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import SwiftUI
+
+/// Implementation detail for `CustomAsyncImage`
+class ImageViewModel: ObservableObject {
+    @Published var state: ImageViewData = .empty
+
+    private var imageClient: ImageClient { Config.imageClient }
```
Comment (SatseAndrew) [inline Sources/SATSCore/BasicComponents/CustomAsyncImage/ImageViewModel.swift:7]: This dependency should be injected in the constructor. 

```suggestion
init(
    state: ImageViewData,
    imageClient: ImageClient = Config.imageClient
) {
    self.state = state
    self.imageClient = imageClient
}
```

**Review by SatseAndrew — APPROVED** [review]
This is a very useful feature. Nice job :) 
I left one comment. It's not a big issue so I will approve this PR instead of requesting changes.

File: Sources/SATSCore/BasicComponents/CustomAsyncImage/ImageViewModel.swift (hunk 2 above)
Comment (fespinoza) [reply Sources/SATSCore/BasicComponents/CustomAsyncImage/ImageViewModel.swift:7]: I did this on purpose.

`ImageViewModel` is a internal type, so the consumer of the library cannot initialize/pass the client along.

`CustomAsyncImage` will create instances of `ImageViewModel`, you can argue that you can inject `ImageClient` to `CustomAsyncImage`, but that would result in a bit of a overly complicated API, as you will use this component inside other components and so on.

For that reason I felt it was more appropriate to inject it to this Singleton proxy object, as Image loading in general is quite a common/generic problem.

Comment (SatseAndrew) [reply Sources/SATSCore/BasicComponents/CustomAsyncImage/ImageViewModel.swift:7]: That sounds fine

**Pull request merged**